tests/Doctrine/Tests/ORM/Functional/ParserResultSerializationTest.php

```diff
@@ -61,6 +64,31 @@ public static function provideSerializedSingleSelectResults(): Generator
         yield '2.15.0' => [rtrim(file_get_contents(__DIR__ . '/ParserResults/single_select_2_15_0.txt'), "\n")];
     }
 
+    public function testSymfony44ProvidedData(): void
+    {
+        $sqlExecutor      = $this->createMock(SingleSelectExecutor::class);
+        $resultSetMapping = $this->createMock(ResultSetMapping::class);
+
+        $tempFile = 'symfony_44-temp.php';
+
+        $parserResult = new ParserResult();
+        $parserResult->setSqlExecutor($sqlExecutor);
+        $parserResult->setResultSetMapping($resultSetMapping);
+        $parserResult->addParameterMapping('name', 0);
+
+        $exported = VarExporter::export($parserResult);
+        $data     = file_put_contents($tempFile, '<?php return ' . $exported . ';');
+
+        $unserialized = require $tempFile;
+
+        $this->assertInstanceOf(ParserResult::class, $unserialized);
+        $this->assertInstanceOf(ResultSetMapping::class, $unserialized->getResultSetMapping());
+        $this->assertEquals(['name' => [0]], $unserialized->getParameterMappings());
+        $this->assertInstanceOf(SingleSelectExecutor::class, $unserialized->getSqlExecutor());
+
+        unlink($tempFile);
```

Comment:
What happens to this file if the test fails?

Reply:
I'm not that famillair with PHPUnit, is there a way to executed something even if the test fails? Maybe i could use `try/finally`?

Reply:
> is there a way to executed something even if the test fails?

Yes, the `tearDown()` method.

Reply:
The `unlink` method is not used after https://github.com/doctrine/orm/pull/10948#discussion_r1387758034